app/src/main/java/ghcharactertracker/com/ghcharactertracker/MainActivity.java

```diff
@@ -0,0 +1,139 @@
+                final AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+                builder.setMessage("Are you sure you want to delete this character?")
+                        .setCancelable(false)
+                        .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int i) {
+                                MainActivity.this.deleteSelectedChar();
+                            }
+                        })
+                        .setNegativeButton("No", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int i) {
+                                dialog.cancel();
+                            }
+                        });
+
+                AlertDialog rUSure = builder.create();
+                rUSure.show();
+
+                MainActivity.this.charToDelete = (Character) charList.getItemAtPosition(position);
+
+                return true;
+            }
+        });
+
+
+        dbHandler = DBHandler.getDbHandler(this);
+        characters = dbHandler.getAllCharacters(characters);
+        adapter.notifyDataSetChanged();
+    }
+
+    public void deleteSelectedChar() {
+        characters.remove(charToDelete);
+        dbHandler.deleteCharacter(charToDelete);
+        adapter.notifyDataSetChanged();
+    }
+
+    public void newCharacter(View v) {
+        Character character = new Character(new CharClass(ClassName.Brute));
+        Scenario scenario = new Scenario();
+        dbHandler.addCharacter(character);
+        dbHandler.addScenario(scenario);
+        character.setCurrentScenario(scenario);
+        startCharacterActivity(character, true);
+    }
+
+    private void startCharacterActivity(Character input, boolean newChar) {
+        Intent characterIntent = new Intent(this, CharacterActivity.class);
+
+        characterIntent.putExtra(CHARACTER_INPUT, input);
+
+        if (newChar) {
+            characterIntent.putExtra(REQUEST_CODE, NEW_CHARACTER_REQUEST_CODE);
+            startActivityForResult(characterIntent, NEW_CHARACTER_REQUEST_CODE);
+        } else {
+            characterIntent.putExtra(REQUEST_CODE, EXISTING_CHARACTER_REQUEST_CODE);
+            startActivityForResult(characterIntent, EXISTING_CHARACTER_REQUEST_CODE);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch (requestCode) {
+            case NEW_CHARACTER_REQUEST_CODE:
+                if (resultCode == Activity.RESULT_OK) { addCharacter(data); }
+                break;
+            case EXISTING_CHARACTER_REQUEST_CODE:
+                if (resultCode == Activity.RESULT_OK) { saveCharacter(data); }
+                break;
```

Comment:
C: might be a good idea to put a default case here that prints some warning. might save you a shitton of time in the future